### PR TITLE
[MODULES] Switch to Go Modules

### LINKED ISSUE #16: arguments_json is not mapped correctly on ReadQueue
### Terraform Version
Terraform v0.11.7
+ provider.rabbitmq v0.2.0

### Affected Resource(s)
- ressource_queue

### Terraform Configuration Files
```hcl
data "template_file" "queue_arguments" {
  template = <<EOF
{
  "x-dead-letter-exchange": "$${exchange}"
}
EOF

  vars {
    exchange = "${rabbitmq_exchange.exchange_dl_.name}"
  }
}

resource "rabbitmq_queue" "queue" {
  count = "${length(var.entities)}"
  name  = "${var.entities[count.index]}.queue"
  vhost = "${rabbitmq_vhost.vhost_1.name}"

  settings {
    durable        = true
    auto_delete    = false
    arguments_json = "${data.template_file.queue_arguments.rendered}"
  }
}
```

### Expected Behavior
The ressource map the arguments as arguments_json and not as arguments.

### Actual Behavior
The refresh map the arguments as arguments therefor, it tries to modify the resources every time with the deletion of the arguments and the application of the arguments_json setting.

### Steps to Reproduce
1. `terraform apply`

### PR DESCRIPTION
**Pull request by appilon**
This will also correct an issue with `arguments_json` in the queue resource. That attribute actually has never worked and is failing silently and causes a permadiff.

Closes #16

### CODE REVIEW

**Review by radeksimko — CHANGES_REQUESTED** [review]
One acceptance test is currently failing on this branch:

```
=== RUN   TestAccQueue_jsonArguments
--- FAIL: TestAccQueue_jsonArguments (0.14s)
    testing.go:538: Step 0 error: Check failed: resource not found: rabbitmq_queue.test
FAIL
```
I'm digging into this.

**Comment by radeksimko** [comment]
I think I found the cause:
```
[ERR] Error writing fields: settings.0.arguments.x-message-ttl: '' expected type 'string', got unconvertible type 'float64'
```

File: rabbitmq/resource_queue.go [hunk 1]
```diff
@@ -60,10 +61,11 @@ func resourceQueue() *schema.Resource {
 						},
 
 						"arguments_json": {
-							Type:          schema.TypeString,
-							Optional:      true,
-							ValidateFunc:  validateJsonString,
-							ConflictsWith: []string{"settings.0.arguments"},
+							Type:             schema.TypeString,
+							Optional:         true,
+							ValidateFunc:     validation.ValidateJsonString,
```
Comment (radeksimko) [inline rabbitmq/resource_queue.go:66]: 👍

File: rabbitmq/resource_queue.go [hunk 2]
```diff
@@ -132,12 +134,28 @@ func ReadQueue(d *schema.ResourceData, meta interface{}) error {
...
+		bytes, err := json.Marshal(queueSettings.Arguments)
+		if err != nil {
+			return err
+		}
+		e["arguments_json"] = string(bytes)
+	} else {
+		e["arguments"] = queueSettings.Arguments
+	}
+
+	queue[0] = e
```
Comment (radeksimko) [inline rabbitmq/resource_queue.go:156]: Nitpick: Do you mind moving the `queue` allocation here above this assignment, just so it's more/immediately obvious there's no risk in addressing the index like this as it is already allocated?

**Review by radeksimko — APPROVED** [review]
This generally LGTM - thanks for chasing it down!

One major suggestion I would have though is to mark `arguments_json` as deprecated, so we can remove it in the future and more importantly make our intentions clear to the user.

**Comment by appilon** [comment]
@radeksimko I will address the allocation thing you mentioned. I think `arguments` would be deprecated as its likely unused in the wild since it cannot support non-string values? Could our decision to deprecate come in a separate PR?

**Comment by apparentlymart** [comment]
Looking forward, I expect in the long run `arguments_json` is what would be deprecated and `arguments` would instead be defined as an arbitrary object rather than a map of string, once the SDK allows that. But as we currently stand, `arguments_json` does indeed seem to be the only way to define non-string arguments, and a string argument with a `_json` suffix is how we've worked around this in a number of other cases, so I think probably best for us to hold on deprecating anything for now until we're ready to address this via an SDK change.

**Pull request merged**